src/index.js

```diff
@@ -6,19 +6,25 @@ export default (router, catchAllHandler) => {
       ...store,
       dispatch(action) {
         const { type, meta } = action;
-        const transition = meta ?
+        const transitionMetaFunc = meta ?
           (meta.transition || catchAllHandler) :
           catchAllHandler;
 
         store.dispatch(action);
 
-        const transitionResult = transition ?
-          transition(store.getState(), action) :
+        const transitionData = transitionMetaFunc ?
+          transitionMetaFunc(store.getState(), action) :
```

Comment:
I renamed these variables because I found the original naming to be very confusing (`transitionResult` sounds like the router transition has already happened to me)